From d453565e2b9ca3cebcbc8a05d93a1dd77d349855 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 11 Jun 2020 07:03:38 +0300
Subject: [PATCH 03/16] Revert "Revert "FODCircleView: dispatch onPress to HAL
 only after dimming is applied""

This reverts commit 0dcd7fee60e41f937c8b52753279f8d70b2e0748.
---
 .../android/systemui/biometrics/FODCircleView.java  | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java b/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
index e480d704a05..6be2fef143d 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
@@ -74,6 +74,8 @@ public class FODCircleView extends ImageView {
     private boolean mIsShowing;
     private boolean mIsCircleShowing;
 
+    private float mCurrentDimAmount = 0.0f;
+
     private Handler mHandler;
 
     private LockPatternUtils mLockPatternUtils;
@@ -192,6 +194,16 @@ public class FODCircleView extends ImageView {
 
         mUpdateMonitor = KeyguardUpdateMonitor.getInstance(context);
         mUpdateMonitor.registerCallback(mMonitorCallback);
+
+        getViewTreeObserver().addOnGlobalLayoutListener(() -> {
+            float drawingDimAmount = mParams.dimAmount;
+            if (mCurrentDimAmount == 0.0f && drawingDimAmount > 0.0f) {
+                dispatchPress();
+                mCurrentDimAmount = drawingDimAmount;
+            } else if (mCurrentDimAmount > 0.0f && drawingDimAmount == 0.0f) {
+                mCurrentDimAmount = drawingDimAmount;
+            }
+        });
     }
 
     @Override
@@ -285,7 +297,6 @@ public class FODCircleView extends ImageView {
 
         setDim(true);
         updateAlpha();
-        dispatchPress();
 
         mPaintFingerprint.setColor(mColor);
         setImageDrawable(null);
-- 
2.17.1

